src/praisonai-agents/praisonaiagents/mcp/hosted_server.py

```
@@ -0,0 +1,190 @@
+"""
+Hosted MCP Server implementation for PraisonAI Agents.
+
+This module provides a base class for creating hosted MCP servers
+that can handle requests and integrate with the MCP protocol.
+"""
+
+import asyncio
+import logging
+from typing import Dict, Any, Optional, List, Callable
+from abc import ABC, abstractmethod
```

Ruff (0.11.9)

`abc.abstractmethod` imported but unused

Remove unused import: `abc.abstractmethod`

(F401)